### ISSUE TITLE
Set PyTorch env vars before importing torch; drop unsupported var

### ISSUE THREAD

**Issue by jashvira** [jashvira/policy-gradient-experiments#1]
Problem
- Env vars for allocator/logging are set after importing torch, so they may not take effect.
- PYTORCH_DISABLE_STACK_TRACE is not a supported Torch env var.

Impact
- Allocator/logging knobs may be ignored; can cause allocator fragmentation and noisy logs.

Code
```python
# experiments/grpo/grpo_train.py
import torch
...
os.environ.setdefault('PYTORCH_CUDA_ALLOC_CONF', 'expandable_segments:True')
os.environ.setdefault('TORCH_LOGS', '-inductor')
os.environ.setdefault('PYTORCH_DISABLE_STACK_TRACE', '1')
```

Proposed fix
- Move Torch-related env var setup above any import torch.
- Remove the unsupported PYTORCH_DISABLE_STACK_TRACE flag.

Acceptance criteria
- Env var block appears before any Torch import.
- No occurrence of PYTORCH_DISABLE_STACK_TRACE in repo.

**Comment by jashvira** [comment]
Fixed by b40c6b9: set PyTorch env vars before importing torch; removed unsupported flag.
